File: scripts/guide-acces-pdf.py
# -*- coding: utf-8 -*-
"""
Le document remis au client après son achat.

Il sert deux fois : MakeTou exige un fichier pour publier un produit, et le
client a besoin de savoir quoi faire s'il ne comprend pas où aller. Le 26 août
2026, treize clients ont payé sans obtenir leur accès et personne ne leur a
rien dit — ce document est le filet qui parle à leur place.

Il redit aussi, noir sur blanc, ce que ProFoot vend : un accès à un outil
d'analyse statistique, payé une fois, sans renouvellement automatique.
"""
from reportlab.lib.pagesizes import A4
from reportlab.lib.units import mm
from reportlab.lib import colors
from reportlab.pdfgen import canvas

VERT = colors.HexColor("#10B981")
ENCRE = colors.HexColor("#0F172A")
GRIS = colors.HexColor("#64748B")
GRIS_CLAIR = colors.HexColor("#E2E8F0")

L, H = A4
M = 22 * mm

c = canvas.Canvas("ProFoot-AI-Guide-Acces.pdf", pagesize=A4)
c.setTitle("ProFoot AI — Comment acceder a votre analyse")
c.setAuthor("ProFoot AI")

# ── Bandeau ────────────────────────────────────────────────────────────────
c.setFillColor(VERT)
c.rect(0, H - 6 * mm, L, 6 * mm, stroke=0, fill=1)

y = H - 30 * mm
c.setFillColor(ENCRE)
c.setFont("Helvetica-Bold", 23)
c.drawString(M, y, "ProFoot AI")

y -= 9 * mm
c.setFillColor(GRIS)
c.setFont("Helvetica", 13)
c.drawString(M, y, "Comment acceder a votre analyse")

y -= 6 * mm
c.setStrokeColor(GRIS_CLAIR)
c.setLineWidth(0.8)
c.line(M, y, L - M, y)

# ── Merci ──────────────────────────────────────────────────────────────────
y -= 12 * mm
c.setFillColor(ENCRE)
c.setFont("Helvetica", 11)
c.drawString(M, y, "Merci pour votre achat. Votre acces est actif des maintenant.")

# ── Les trois etapes ───────────────────────────────────────────────────────
etapes = [
    ("1", "Rendez-vous sur profootai.com",
     "Depuis un telephone ou un ordinateur, avec votre navigateur habituel."),
    ("2", "Connectez-vous avec l'adresse e-mail utilisee pour l'achat",
     "C'est elle qui relie votre paiement a votre compte. Une adresse",
     "differente ne trouvera pas votre acces."),
    ("3", "Votre acces s'ouvre automatiquement",
     "Aucun code a saisir, aucune manipulation. Lancez votre premiere",
     "analyse en choisissant deux equipes."),
]

y -= 14 * mm
for e in etapes:
    numero, titre = e[0], e[1]
    lignes = e[2:]

    c.setFillColor(VERT)
    c.circle(M + 3.5 * mm, y + 1.4 * mm, 3.5 * mm, stroke=0, fill=1)
    c.setFillColor(colors.white)
    c.setFont("Helvetica-Bold", 10)
    c.drawCentredString(M + 3.5 * mm, y - 0.4 * mm, numero)

    c.setFillColor(ENCRE)
    c.setFont("Helvetica-Bold", 11.5)
    c.drawString(M + 12 * mm, y, titre)

    yy = y - 5.6 * mm
    c.setFillColor(GRIS)
    c.setFont("Helvetica", 10)
    for ligne in lignes:
        c.drawString(M + 12 * mm, yy, ligne)
        yy -= 4.8 * mm

    y = yy - 6 * mm

# ── Ce que comprend l'acces ────────────────────────────────────────────────
y -= 2 * mm
c.setFillColor(colors.HexColor("#F0FDF4"))
c.roundRect(M, y - 34 * mm, L - 2 * M, 34 * mm, 3 * mm, stroke=0, fill=1)

c.setFillColor(ENCRE)
c.setFont("Helvetica-Bold", 11)
c.drawString(M + 7 * mm, y - 9 * mm, "Ce que comprend votre acces Essentiel")

c.setFillColor(GRIS)
c.setFont("Helvetica", 10)
for i, ligne in enumerate([
    "20 analyses completes par mois, sur plus de 15 competitions",
    "Statistiques avancees : forme, confrontations, buts attendus",
    "Acces valable 30 jours a compter de votre achat",
]):
    c.setFillColor(VERT)
    c.setFont("Helvetica-Bold", 10)
    c.drawString(M + 7 * mm, y - 16 * mm - i * 5.5 * mm, "-")
    c.setFillColor(GRIS)
    c.setFont("Helvetica", 10)
    c.drawString(M + 11 * mm, y - 16 * mm - i * 5.5 * mm, ligne)

y -= 44 * mm

# ── Un souci ? ─────────────────────────────────────────────────────────────
c.setFillColor(ENCRE)
c.setFont("Helvetica-Bold", 11)
c.drawString(M, y, "Un probleme pour acceder ?")

y -= 6 * mm
c.setFillColor(GRIS)
c.setFont("Helvetica", 10)
for ligne in [
    "Ecrivez-nous a contactprofootai@gmail.com en precisant l'adresse e-mail",
    "utilisee lors de l'achat. Nous ouvrons votre acces manuellement.",
]:
    c.drawString(M, y, ligne)
    y -= 5 * mm

# ── Pied de page ───────────────────────────────────────────────────────────
y = 24 * mm
c.setStrokeColor(GRIS_CLAIR)
c.line(M, y + 8 * mm, L - M, y + 8 * mm)

c.setFillColor(GRIS)
c.setFont("Helvetica", 8.5)
for ligne in [
    "ProFoot AI est un outil d'analyse statistique du football. L'achat donne acces a l'outil pour une duree",
    "determinee : paiement unique, sans renouvellement automatique. Aucune analyse ne garantit un resultat.",
]:
    c.drawString(M, y, ligne)
    y -= 4.5 * mm

c.setFillColor(VERT)
c.setFont("Helvetica-Bold", 9)
c.drawString(M, y - 2 * mm, "profootai.com")

c.showPage()
c.save()
print("  PDF cree : ProFoot-AI-Guide-Acces.pdf")

File: scripts/guide-acces-pdf.mjs
/**
 * Le document remis au client après son achat.
 *
 * Il sert deux fois : MakeTou exige un fichier pour publier un produit, et le
 * client a besoin de savoir quoi faire s'il ne comprend pas où aller. Le 26 août
 * 2026, treize clients ont payé sans obtenir leur accès et personne ne leur a
 * rien dit — ce document est le filet qui parle à leur place.
 *
 * Il redit aussi, noir sur blanc, ce que ProFoot vend : un accès à un outil
 * d'analyse statistique, payé une fois, sans renouvellement automatique.
 */
import fs from 'node:fs';
import PDFDocument from 'pdfkit';

const VERT = '#10B981';
const VERT_PALE = '#F0FDF4';
const ENCRE = '#0F172A';
const GRIS = '#64748B';
const GRIS_CLAIR = '#E2E8F0';

const doc = new PDFDocument({ size: 'A4', margin: 0, info: {
  Title: 'ProFoot AI — Comment accéder à votre analyse',
  Author: 'ProFoot AI',
} });
doc.pipe(fs.createWriteStream('ProFoot-AI-Guide-Acces.pdf'));

const L = doc.page.width;
const M = 62;

// ── Bandeau ────────────────────────────────────────────────────────────────
doc.rect(0, 0, L, 16).fill(VERT);

let y = 78;
doc.fillColor(ENCRE).font('Helvetica-Bold').fontSize(24).text('ProFoot AI', M, y);

y += 30;
doc.fillColor(GRIS).font('Helvetica').fontSize(13.5)
   .text('Comment accéder à votre analyse', M, y);

y += 26;
doc.moveTo(M, y).lineTo(L - M, y).lineWidth(0.8).stroke(GRIS_CLAIR);

// ── Merci ──────────────────────────────────────────────────────────────────
y += 26;
doc.fillColor(ENCRE).font('Helvetica').fontSize(11)
   .text('Merci pour votre achat. Votre accès est actif dès maintenant.', M, y);

// ── Les trois étapes ───────────────────────────────────────────────────────
const etapes = [
  ['1', 'Rendez-vous sur profootai.com',
   ["Depuis un téléphone ou un ordinateur, avec votre navigateur habituel."]],
  ['2', "Connectez-vous avec l'adresse e-mail utilisée pour l'achat",
   ["C'est elle qui relie votre paiement à votre compte. Une adresse",
    "différente ne trouvera pas votre accès."]],
  ['3', "Votre accès s'ouvre automatiquement",
   ['Aucun code à saisir, aucune manipulation. Lancez votre première',
    'analyse en choisissant deux équipes.']],
];

y += 36;
for (const [numero, titre, lignes] of etapes) {
  doc.circle(M + 10, y + 6, 10).fill(VERT);
  doc.fillColor('#FFFFFF').font('Helvetica-Bold').fontSize(10.5)
     .text(numero, M + 5, y + 2, { width: 10, align: 'center' });

  doc.fillColor(ENCRE).font('Helvetica-Bold').fontSize(11.5)
     .text(titre, M + 32, y);

  let yy = y + 17;
  doc.fillColor(GRIS).font('Helvetica').fontSize(10);
  for (const ligne of lignes) {
    doc.text(ligne, M + 32, yy);
    yy += 14;
  }
  y = yy + 16;
}

// ── Ce que comprend l'accès ────────────────────────────────────────────────
const hEncadre = 104;
doc.roundedRect(M, y, L - 2 * M, hEncadre, 8).fill(VERT_PALE);

doc.fillColor(ENCRE).font('Helvetica-Bold').fontSize(11)
   .text('Ce que comprend votre accès Essentiel', M + 20, y + 18);

const inclus = [
  '20 analyses complètes par mois, sur plus de 15 compétitions',
  'Statistiques avancées : forme, confrontations, buts attendus',
  'Accès valable 30 jours à compter de votre achat',
];
inclus.forEach((ligne, i) => {
  const yl = y + 42 + i * 17;
  doc.fillColor(VERT).font('Helvetica-Bold').fontSize(10).text('•', M + 20, yl);
  doc.fillColor(GRIS).font('Helvetica').fontSize(10).text(ligne, M + 32, yl);
});

y += hEncadre + 30;

// ── Un souci ? ─────────────────────────────────────────────────────────────
doc.fillColor(ENCRE).font('Helvetica-Bold').fontSize(11)
   .text('Un problème pour accéder ?', M, y);

y += 18;
doc.fillColor(GRIS).font('Helvetica').fontSize(10);
for (const ligne of [
  "Écrivez-nous à contactprofootai@gmail.com en précisant l'adresse e-mail",
  'utilisée lors de votre achat. Nous ouvrons votre accès manuellement.',
]) {
  doc.text(ligne, M, y);
  y += 14;
}

// ── Pied de page ───────────────────────────────────────────────────────────
const yPied = doc.page.height - 92;
doc.moveTo(M, yPied).lineTo(L - M, yPied).lineWidth(0.8).stroke(GRIS_CLAIR);

doc.fillColor(GRIS).font('Helvetica').fontSize(8.5);
let yp = yPied + 14;
for (const ligne of [
  "ProFoot AI est un outil d'analyse statistique du football. L'achat donne accès à l'outil pour une durée",
  'déterminée : paiement unique, sans renouvellement automatique. Aucune analyse ne garantit un résultat.',
]) {
  doc.text(ligne, M, yp);
  yp += 12;
}

doc.fillColor(VERT).font('Helvetica-Bold').fontSize(9).text('profootai.com', M, yp + 6);

doc.end();
console.log('  PDF créé : ProFoot-AI-Guide-Acces.pdf');
